Sparkle/SUOperatingSystem.m

```diff
@@ -20,11 +20,14 @@ + (NSOperatingSystemVersion)operatingSystemVersion
     {
         NSOperatingSystemVersion version = { 0, 0, 0 };
         NSURL *coreServices = [[NSFileManager defaultManager] URLForDirectory:NSCoreServiceDirectory inDomain:NSSystemDomainMask appropriateForURL:nil create:NO error:nil];
-        NSDictionary *dictionary = [NSDictionary dictionaryWithContentsOfURL:[coreServices URLByAppendingPathComponent:@"SystemVersion.plist"]];
-        NSArray *components = [ [dictionary objectForKey: @"ProductVersion"] componentsSeparatedByString:@"."];
-        version.majorVersion = components.count > 0 ? [ [components objectAtIndex:0] integerValue] : 0;
-        version.minorVersion = components.count > 1 ? [ [components objectAtIndex:1] integerValue] : 0;
-        version.patchVersion = components.count > 2 ? [ [components objectAtIndex:2] integerValue] : 0;
+        NSURL *url = [coreServices URLByAppendingPathComponent:@"SystemVersion.plist"] ;
+        if (url) {
```

Comment:
I think we should assert(url != nil) instead of using an if, since if this fails, the logic in the calling code is going to be broken..